### PR TITLE
Authenticode sign file in Linux build

### LINKED ISSUE #18247: Powershell files from official Microsoft Nuget Microsoft.PowerShell.SDK are not signed
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Some psd1 files, which are a part of official Microsoft nuget package - [NuGet Gallery | Microsoft.PowerShell.SDK 7.2.6](https://www.nuget.org/packages/Microsoft.PowerShell.SDK/) are not signed. Specifically, all the files in `contentFiles/any/any/runtimes/unix` are missing signatures.

We are an internal MS team depending on this Nuget and are experiencing build failures due to missing signatures in the mentioned files.

### Expected behavior

```console
The files should be signed
```

### Actual behavior

```console
The files are not signed
```

### Error details

_No response_

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.6
PSEdition                      Core
GitCommitId                    7.2.6
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Autheticode sign all files in linux builds

## PR Context

Fixes #18247

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/releaseBuild/azureDevOps/templates/linux.yml [hunk 1]
```diff
@@ -84,6 +103,24 @@ jobs:
...
+        pattern: |
+          **\*.dll
+          **\*.psd1
+          **\*.psm1
+          **\*.ps1xml
+          **\*.ps1
+          **\*.exe
+        useMinimatch: true
+        shouldSign: $(SHOULD_SIGN)
+        displayName: Sign Linux Framework Dependent
```
Comment (TravisEz13) [inline tools/releaseBuild/azureDevOps/templates/linux.yml:121]:
```suggestion
        displayName: Sign Linux Mariner x64
```

**Review by TravisEz13 — APPROVED** [review]

File: tools/releaseBuild/azureDevOps/templates/linux.yml [hunk 2]
```diff
@@ -94,6 +131,24 @@ jobs:
...
         sourceScanPath: '$(PowerShellRoot)/tools'
 
+  - ${{ if eq(variables.build,'deb') }} :
+    - template: EsrpSign.yml@ComplianceRepo
+      parameters:
+        buildOutputPath: '$(System.ArtifactsDirectory)/pwshLinuxBuildMinSize'
+        signOutputPath: '$(System.ArtifactsDirectory)/pwshLinuxBuildMinSize'
+        certificateId: "CP-230012"
+        pattern: |
+          **\*.dll
```
Comment (daxian-dbw) [inline tools/releaseBuild/azureDevOps/templates/linux.yml:141]: Do they request for signing DLLs too?

File: tools/releaseBuild/azureDevOps/templates/linux.yml [hunk 3]
```diff
@@ -94,6 +131,24 @@ jobs:
...
+          **\*.psd1
+          **\*.psm1
+          **\*.ps1xml
+          **\*.ps1
+          **\*.exe
```
Comment (daxian-dbw) [inline tools/releaseBuild/azureDevOps/templates/linux.yml:146]: There is no `*.exe` files in Linux build.

File: tools/releaseBuild/azureDevOps/templates/linux.yml (hunk 2 above)
Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/templates/linux.yml:141]: All files that can be embedded signed are required to be signed

File: tools/releaseBuild/azureDevOps/templates/linux.yml (hunk 3 above)
Comment (TravisEz13) [reply tools/releaseBuild/azureDevOps/templates/linux.yml:146]:
```suggestion
```
ELF files cannot be authenticode signed so there is nothing to replace this with.

**Comment by daxian-dbw** [comment]
@adityapatwardhan Please feel free to merge this PR when it's ready.

**Comment by adityapatwardhan** [comment]
This is draft.. More changes are coming.

**Pull request closed without merging**